psydac/linalg/stencil.py

```diff
@@ -570,14 +613,15 @@ def __init__( self, V, W, pads=None , backend=None):
         if pads is not None:
             for p,vp in zip(pads, V.pads):
                 assert p<=vp
-        
+
         self._pads     = pads or tuple(V.pads)
-        dims           = [e-s+2*p+1 for s,e,p in zip(W.starts, W.ends, W.pads)]
-        diags          = [2*p+1 for p in self._pads]
+        dims           = [e-s+2*mi*p+1 for s,e,p,mi in zip(W.starts, W.ends, W.pads, W.multiplicity)]
+        diags          = [compute_diag_len(p, md, mc) for p,md,mc in zip(self._pads, V.multiplicity, W.multiplicity)]
```

Comment:
As I recommended in a previous comment, here we can pass the multiplicities of domain and codomain as arguments to the constructor of `StencilMatrix` instead of storing them inside the `StencilVectorSpace` objects `V` and `M`.

Reply:
we discussed this previously, the information is part of the space, we need that in the decomposition